Flaky test: test_library_bridge/test.py::test_server_restart_bridge_might_be_still_alive

_Important: This issue was automatically generated and is used by CI for matching failures. DO NOT modify the body content. DO NOT remove labels._

Test name: test_library_bridge/test.py::test_server_restart_bridge_might_be_still_alive
Failure reason: Code:
CI report: [Integration tests (amd_tsan, 4/6)](https://s3.amazonaws.com/clickhouse-test-reports/json.html?PR=105337&sha=fa5f98d9e1b6c17f0963df31f77056751ee94b0f&name_0=PR&name_1=Integration%20tests%20%28amd_tsan%2C%204%2F6%29)

Failing test history: [cidb](https://play.clickhouse.com#V0lUSAogICAgOTAgQVMgaW50ZXJ2YWxfZGF5cwpTRUxFQ1QKICAgIHRvU3RhcnRPZkRheShjaGVja19zdGFydF90aW1lKSBBUyBkYXksCiAgICBjb3VudCgpIEFTIGZhaWx1cmVzLAogICAgZ3JvdXBVbmlxQXJyYXkocHVsbF9yZXF1ZXN0X251bWJlcikgQVMgcHJzLAogICAgYW55KHJlcG9ydF91cmwpIEFTIHJlcG9ydF91cmwKRlJPTSBjaGVja3MKV0hFUkUgKG5vdygpIC0gdG9JbnRlcnZhbERheShpbnRlcnZhbF9kYXlzKSkgPD0gY2hlY2tfc3RhcnRfdGltZQogICAgQU5EIHRlc3RfbmFtZSA9ICd0ZXN0X2xpYnJhcnlfYnJpZGdlL3Rlc3QucHk6OnRlc3Rfc2VydmVyX3Jlc3RhcnRfYnJpZGdlX21pZ2h0X2JlX3N0aWxsX2FsaXZlJwogICAgLS0gQU5EIGNoZWNrX25hbWUgPSAnSW50ZWdyYXRpb24gdGVzdHMgKGFtZF90c2FuLCA0LzYpJwogICAgQU5EIHRlc3Rfc3RhdHVzIElOICgnRkFJTCcsICdFUlJPUicpCiAgICBBTkQgKHB1bGxfcmVxdWVzdF9udW1iZXIgPSAwIE9SIGJhc2VfcmVmIElOICgnbWFzdGVyJykpCiAgICBBTkQgdGVzdF9jb250ZXh0X3JhdyBMSUtFICclQ29kZTolJwpHUk9VUCBCWSBkYXkKT1JERVIgQlkgZGF5IERFU0MK)

Test output:
```
File: test_library_bridge/test.py:307 - in test_server_restart_bridge_might_be_still_alive
    result = instance.query("""select dictGet(lib_dict_c, 'value1', toUInt64(1));""")
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File: helpers/cluster.py:4933 - in query
    return self.client.query(
File: helpers/client.py:40 - in wrap
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
File: helpers/client.py:80 - in query
    ).get_answer()
      ^^^^^^^^^^^^
File: helpers/client.py:269 - in get_answer
    raise QueryRuntimeException(
E   helpers.client.QueryRuntimeException: Client failed! Return code: 254, stderr: Received exception from server (version 26.6.1):
E   Code: 510. DB::Exception: Received from 172.16.1.2:9000. DB::Exception: Update failed for dictionary 'default.lib_dict_c (c1ed087f-e0c0-4042-868e-d7453280ab08)': Code: 510. DB::Exception: Update failed for dictionary default.lib_dict_c (c1ed087f-e0c0-4042-868e-d7453280ab08) : Poco::Exception. Code: 1000, e.code() = 104, Connection reset by peer, Stack trace (when copying this message, always include the lines below):
E   
E   0. contrib/llvm-project/libcxx/include/__exception/exception.h:115: Poco::IOException::IOException(int) @ 0x000000003758dfda
E   1. base/poco/Net/src/NetException.cpp:26:1: Poco::Net::ConnectionResetException::ConnectionResetException(int) @ 0x0000000037696e21
E   2. base/poco/Net/src/SocketImpl.cpp:1083:9: Poco::Net::SocketImpl::error(int, String const&) @ 0x00000000376aea36
E   3. base/poco/Net/src/SocketImpl.cpp:1021:2: Poco::Net::SocketImpl::receiveBytes(void*, int, int) @ 0x00000000376b1928
E   4. base/poco/Net/src/StreamSocket.cpp:135:17: Poco::Net::StreamSocket::receiveBytes(void*, int, int) @ 0x00000000376b8acb
E   5. base/poco/Net/src/HTTPSession.cpp:204:24: Poco::Net::HTTPSession::receive(char*, int) @ 0x0000000037676da1
E   6. base/poco/Net/src/HTTPSession.cpp:173:14: Poco::Net::HTTPSession::read(char*, long) @ 0x0000000037676cfe
E   7. base/poco/Net/src/HTTPStream.cpp:54:18: Poco::Net::HTTPStreamBuf::readFromDevice(char*, long) @ 0x0000000037679d40
E   8. src/IO/ReadBufferFromIStream.cpp:27:64: DB::ReadBufferFromIStream::nextImpl() @ 0x000000001c3a25dd
E   9. src/IO/ReadBuffer.cpp:96:15: DB::ReadBuffer::next() @ 0x00000000183da304
E   10. src/IO/ReadWriteBufferFromHTTP.cpp:511:33: void std::__function::__policy_func<void ()>::__call_func[abi:fqe220101]<DB::ReadWriteBufferFromHTTP::nextImpl()::$_0>(std::__function::__policy_storage const*) @ 0x000000001c39e1a8
E   11. contrib/llvm-project/libcxx/include/__functional/function.h:502:12: ? @ 0x000000001c3956cb
E   12. src/IO/ReadWriteBufferFromHTTP.cpp:471:5: DB::ReadWriteBufferFromHTTP::nextImpl() @ 0x000000001c39b42c
E   13. src/IO/ReadBuffer.cpp:96:15: DB::ReadBuffer::next() @ 0x00000000183da304
E   14. src/IO/ReadBuffer.h:81:38: DB::BinaryRowInputFormat<false>::readRow(std::vector<COW<DB::IColumn>::mutable_ptr<DB::IColumn>, std::allocator<COW<DB::IColumn>::mutable_ptr<DB::IColumn>>>&, DB::RowReadExtension&) @ 0x000000002c44dc6a
E   15. src/Processors/Formats/IRowInputFormat.cpp:177:36: DB::IRowInputFormat::read() @ 0x000000002c356793
E   16. src/Processors/Formats/IInputFormat.cpp:33:21: DB::IInputFormat::generate() @ 0x000000002c34ff77
E   17. src/Processors/ISource.cpp:153:18: DB::ISource::tryGenerate() @ 0x000000002c3092f8
E   18. src/Processors/ISource.cpp:119:26: DB::ISource::work() @ 0x000000002c308dae
E   19. src/Processors/Executors/ExecutionThreadContext.cpp:54:28: DB::ExecutionThreadContext::executeTask() @ 0x000000002c339d73
E   20. src/Processors/Executors/PipelineExecutor.cpp:363:26: DB::PipelineExecutor::executeStepImpl(unsigned long, DB::IAcquiredSlot*, std::atomic<bool>*) @ 0x000000002c3223d9
E   21. src/Processors/Executors/PipelineExecutor.cpp:191:5: DB::PipelineExecutor::executeStep(std::atomic<bool>*) @ 0x000000002c321d96
E   22. src/Processors/Executors/PullingPipelineExecutor.cpp:60:20: DB::PullingPipelineExecutor::pull(DB::Chunk&) @ 0x000000002c347d67
E   23. src/Processors/Executors/PullingPipelineExecutor.cpp:71:10: DB::PullingPipelineExecutor::pull(DB::Block&) @ 0x000000002c347fc1
E   24. src/Dictionaries/DictionaryPipelineExecutor.cpp:36:26: DB::DictionaryPipelineExecutor::pull(DB::Block&) @ 0x000000001c9f9551
E   25. src/Dictionaries/CacheDictionary.cpp:653:33: DB::CacheDictionary<(DB::DictionaryKeyType)0>::update(std::shared_ptr<DB::CacheDictionaryUpdateUnit<(DB::DictionaryKeyType)0>>)::'lambda'()::operator()() const @ 0x000000001ccbf36d
E   26. src/QueryPipeline/BlockIO.h:75:13: void DB::BlockIO::executeWithCallbacks<DB::CacheDictionary<(DB::DictionaryKeyType)0>::update(std::shared_ptr<DB::CacheDictionaryUpdateUnit<(DB::DictionaryKeyType)0>>)::'lambda'()>(DB::CacheDictionary<(DB::DictionaryKeyType)0>::update(std::shared_ptr<DB::CacheDictionaryUpdateUnit<(DB::DictionaryKeyType)0>>)::'lambda'()&&) @ 0x000000001ccb1e5e
E   27. src/Dictionaries/CacheDictionary.cpp:647:16: DB::CacheDictionary<(DB::DictionaryKeyType)0>::update(std::shared_ptr<DB::CacheDictionaryUpdateUnit<(DB::DictionaryKeyType)0>>) @ 0x000000001ccb0347
E   28. src/Dictionaries/CacheDictionary.cpp:61:13: void std::__function::__policy_func<void (std::shared_ptr<DB::CacheDictionaryUpdateUnit<(DB::DictionaryKeyType)0>>)>::__call_func[abi:fqe220101]<DB::CacheDictionary<(DB::DictionaryKeyType)0>::CacheDictionary(DB::StorageID const&, DB::DictionaryStructure const&, std::shared_ptr<DB::IDictionarySource>, std::shared_ptr<DB::ICacheDictionaryStorage>, DB::CacheDictionaryUpdateQueueConfiguration, DB::CacheDictionaryConfiguration)::'lambda'(std::shared_ptr<DB::CacheDictionaryUpdateUnit<(DB::DictionaryKeyType)0>>)>(std::__function::__policy_storage const*, std::shared_ptr<DB::CacheDictionaryUpdateUnit<(DB::DictionaryKeyType)0>>&&) @ 0x000000001ccbe890
E   29. contrib/llvm-project/libcxx/include/__functional/function.h:502:12: ? @ 0x000000001ccc5aa2
E   30. src/Dictionaries/CacheDictionaryUpdateQueue.cpp:40:57: void std::__function::__policy_func<void ()>::__call_func[abi:fqe220101]<DB::CacheDictionaryUpdateQueue<(DB::DictionaryKeyType)0>::CacheDictionaryUpdateQueue(String, DB::CacheDictionaryUpdateQueueConfiguration, std::function<void (std::shared_ptr<DB::CacheDictionaryUpdateUnit<(DB::DictionaryKeyType)0>>)>&&)::'lambda'()>(std::__function::__policy_storage const*) @ 0x000000001ccc78a2
E   31. contrib/llvm-project/libcxx/include/__functional/function.h:502:12: ? @ 0x000000001847e510
E    (version 26.6.1.226). (CACHE_DICTIONARY_UPDATE_FAIL) (version 26.6.1.226): In scope SELECT dictGet(lib_dict_c, 'value1', toUInt64(1)). Stack trace:
E   
~~~~~ truncated 34 lines ~~~~~
```

<!-- ch-version-info:start -->
### Version info
- Resolved by: #106152
- Merged into: `26.6.1.276`
<!-- ch-version-info:end -->

--- a/tests/integration/test_library_bridge/test.py
+++ b/tests/integration/test_library_bridge/test.py
@@ -303,6 +303,9 @@ def test_server_restart_bridge_might_be_still_alive(ch_cluster):
     assert result.strip() == "101"
 
     instance.restart_clickhouse()
+    # Force the server to re-establish the bridge connection; the old bridge process
+    # may still be alive with stale state, causing "Connection reset by peer" on first dictGet.
+    instance.query("SYSTEM RELOAD DICTIONARY lib_dict_c")
 
     result = instance.query("""select dictGet(lib_dict_c, 'value1', toUInt64(1));""")
     assert result.strip() == "101"
@@ -311,6 +314,7 @@ def test_server_restart_bridge_might_be_still_alive(ch_cluster):
         ["bash", "-c", "kill -9 `pidof clickhouse-library-bridge`"], user="root"
     )
     instance.restart_clickhouse()
+    instance.query("SYSTEM RELOAD DICTIONARY lib_dict_c")
 
     result = instance.query("""select dictGet(lib_dict_c, 'value1', toUInt64(1));""")
     assert result.strip() == "101"

